ipfs-cluster-service/main.go

```diff
@@ -337,12 +352,33 @@ func setupDebug() {
 	//SetFacilityLogLevel("libp2p-raft", l)
 }
 
-func initConfig(force bool) {
+func initConfig(force bool, generateSecret bool, envSecret string) {
 	if _, err := os.Stat(configPath); err == nil && !force {
 		err := fmt.Errorf("%s exists. Try running with -f", configPath)
 		checkErr("", err)
 	}
-	cfg, err := ipfscluster.NewDefaultConfig()
+
+	var swarmSecret *string
+	if generateSecret {
+		// we will generate the secret automatically
+		swarmSecret = nil
+	} else if envSecret != "" {
+		// read swarm secret from env variable
+		fmt.Println("Reading swarm secret from SWARM_SECRET environment variable.")
+		swarmSecret = &envSecret
+	} else {
+		// get swarm secret from user
+		scanner := bufio.NewScanner(os.Stdin)
+		fmt.Print("Enter swarm secret (to automatically generate, rerun with --generate-secret): ")
+		scanner.Scan()
+		// QuoteToASCII leaves outer double quotes on the resulting string,
+		// Unqoute removes them, and hence there *should* never be an error, so
+		// we ignore it
+		input, _ := strconv.Unquote(strconv.QuoteToASCII(scanner.Text()))
```

Comment:
not sure this is needed. in the end the string should be just casted to []byte, and go handles that well for non-ascii chars.